### PR TITLE
Auth: Cleanup unused argument warnings, and other things

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
Building with `-Wall -Wextra -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int` throws a lot of warnings, mostly about unused arguments. A lot of them are coming from header files which tend to spam the compiler output, making it easier to miss more important ones. This PR fixes most of these build-time warnings.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/cachecleaner.hh [hunk 1]
```diff
@@ -31,7 +31,7 @@
 // the ritual is that the oldest entries are in *front* of the sequence collection, so on a hit, move an item to the end
 // and optionally, on a miss, move it to the beginning
 template <typename S, typename C, typename T>
-void pruneCollection(C& container, T& collection, size_t maxCached, size_t scanFraction = 1000)
+void pruneCollection(C& /* container */, T& collection, size_t maxCached, size_t scanFraction = 1000)
```
Comment (rgacogne) [inline pdns/cachecleaner.hh:34]: It looks like the Container parameter (and template type) could just go?

File: pdns/dbdnsseckeeper.cc [hunk 2]
```diff
@@ -682,7 +682,7 @@ bool DNSSECKeeper::getTSIGForAccess(const DNSName& zone, const ComboAddress& mas
   return false;
 }
 
-bool DNSSECKeeper::unSecureZone(const DNSName& zone, string& error, string& info) {
+bool DNSSECKeeper::unSecureZone(const DNSName& zone, string& error, string& /* info */) {
```
Comment (rgacogne) [inline pdns/dbdnsseckeeper.cc:685]: We should remove the `info` parameter.

File: pdns/dnsbackend.hh [hunk 3]
```diff
@@ -50,22 +50,22 @@ struct SOAData;
...
-  uint32_t id;
-  uint32_t notified_serial;
+  uint32_t id{};
+  uint32_t notified_serial{};
 
-  bool receivedNotify;
+  bool receivedNotify{};
 
-  uint32_t serial;
+  uint32_t serial{};
```
Comment (rgacogne) [inline pdns/dnsbackend.hh:68]: It's a matter of taste so we probably need to discuss this with the team, but I really prefer using explicit values (`0`, `nullptr`, `0`, `0`, `false`, `0`).

File: pdns/dnsrecords.cc [hunk 4]
```diff
@@ -886,7 +888,7 @@ bool getEDNSOpts(const MOADNSParser& mdp, EDNSOpts* eo)
   return false;
 }
 
-DNSRecord makeOpt(const uint16_t udpsize, const uint16_t extRCode, const uint16_t extFlags)
+DNSRecord makeOpt(const uint16_t udpsize, const uint16_t /* extRCode */, const uint16_t extFlags)
```
Comment (rgacogne) [inline pdns/dnsrecords.cc:891]: This looks like a bug, but the function seems unused so we might be able to just remove it?

File: pdns/pdnsutil.cc [hunk 5]
```diff
@@ -1120,7 +1120,7 @@ static int read1char(){
     return c;
 }
 
-static int clearZone(DNSSECKeeper& dk, const DNSName &zone) {
+static int clearZone(DNSSECKeeper& /* dk */, const DNSName &zone) {
```
Comment (rgacogne) [inline pdns/pdnsutil.cc:1123]: The parameter can go.

File: pdns/pdnsutil.cc [hunk 6]
```diff
@@ -1785,7 +1785,7 @@ static bool testAlgorithms()
   return DNSCryptoKeyEngine::testAll();
 }
 
-static void testSpeed(DNSSECKeeper& dk, const DNSName& zone, const string& remote, int cores)
+static void testSpeed(DNSSECKeeper& /* dk */, const DNSName& zone, const string& /* remote */, int cores)
```
Comment (rgacogne) [inline pdns/pdnsutil.cc:1788]: The parameter can go.

File: pdns/pkcs11signers.hh [hunk 7]
```diff
@@ -34,7 +34,7 @@ class PKCS11DNSCryptoKeyEngine : public DNSCryptoKeyEngine
     PKCS11DNSCryptoKeyEngine(unsigned int algorithm);
     ~PKCS11DNSCryptoKeyEngine();
 
-    bool operator<(const PKCS11DNSCryptoKeyEngine& rhs) const
+    bool operator<(const PKCS11DNSCryptoKeyEngine& /* rhs */) const
```
Comment (rgacogne) [inline pdns/pkcs11signers.hh:37]: This is scary, but hopefully unused, we might be able to remove it?

File: pdns/slavecommunicator.cc [hunk 8]
```diff
@@ -421,8 +421,7 @@ static bool catalogProcess(const DomainInfo& di, vector<DNSResourceRecord>& rrs,
   return catalogDiff(di, fromXFR, fromDB, logPrefix);
 }
 
-void CommunicatorClass::ixfrSuck(const DNSName &domain, const TSIGTriplet& tt, const ComboAddress& laddr, const ComboAddress& remote, unique_ptr<AuthLua4>& pdl,
-                                 ZoneStatus& zs, vector<DNSRecord>* axfr)
+void CommunicatorClass::ixfrSuck(const DNSName& domain, const TSIGTriplet& tt, const ComboAddress& laddr, const ComboAddress& remote, unique_ptr<AuthLua4>& /* pdl */, ZoneStatus& zs, vector<DNSRecord>* axfr)
```
Comment (rgacogne) [inline pdns/slavecommunicator.cc:424]: Looks like the parameter can go.

File: pdns/sstuff.hh [hunk 9]
```diff
@@ -180,11 +180,11 @@ public:
     d_buffer.resize(s_buflen);
     if((bytes=recvfrom(d_socket, &d_buffer[0], s_buflen, 0, reinterpret_cast<sockaddr *>(&ep) , &remlen)) <0)
       throw NetworkError("After recvfrom: "+stringerror());
-    
+
     dgram.assign(d_buffer, 0, static_cast<size_t>(bytes));
   }
 
-  bool recvFromAsync(string &dgram, ComboAddress &ep)
+  bool recvFromAsync(string &dgram, ComboAddress& /* ep */)
```
Comment (rgacogne) [inline pdns/sstuff.hh:187]: We can remove this parameter.

File: pdns/tcpiohandler.hh [hunk 10]
```diff
@@ -233,7 +233,7 @@ class TCPIOHandler
 public:
   enum class Type : uint8_t { Client, Server };
 
-  TCPIOHandler(const std::string& host, bool hostIsAddr, int socket, const struct timeval& timeout, std::shared_ptr<TLSCtx> ctx, time_t now): d_socket(socket)
+  TCPIOHandler(const std::string& host, bool hostIsAddr, int socket, const struct timeval& timeout, std::shared_ptr<TLSCtx> ctx, time_t /* now */): d_socket(socket)
```
Comment (rgacogne) [inline pdns/tcpiohandler.hh:236]: We should remove this parameter.

File: pdns/ws-auth.cc [hunk 11]
```diff
@@ -536,7 +536,7 @@ static void validateGatheredRRType(const DNSResourceRecord& rr) {
   }
 }
 
-static void gatherRecords(UeberBackend& B, const string& logprefix, const Json& container, const DNSName& qname, const QType& qtype, const int ttl, vector<DNSResourceRecord>& new_records) {
+static void gatherRecords(UeberBackend& /* B */, const string& /* logprefix */, const Json& container, const DNSName& qname, const QType& qtype, const int ttl, vector<DNSResourceRecord>& new_records) {
```
Comment (rgacogne) [inline pdns/ws-auth.cc:539]: The parameters should go.

File: pdns/zone2json.cc [hunk 12]
```diff
@@ -53,7 +53,7 @@ using namespace json11;
 StatBag S;
 static int g_numRecords;
 
-static Json::object emitRecord(const string& zoneName, const DNSName &DNSqname, const string &qtype, const string &ocontent, int ttl)
+static Json::object emitRecord(const string& /* zoneName */, const DNSName &DNSqname, const string &qtype, const string &ocontent, int ttl)
```
Comment (rgacogne) [inline pdns/zone2json.cc:56]: This parameter should go.

File: modules/pipebackend/coprocess.cc [hunk 13]
```diff
@@ -218,7 +218,7 @@ void CoProcess::sendReceive(const string& snd, string& rcv)
   receive(rcv);
 }
 
-UnixRemote::UnixRemote(const string& path, int timeout)
+UnixRemote::UnixRemote(const string& path, int /* timeout */)
```
Comment (rgacogne) [inline modules/pipebackend/coprocess.cc:221]: Looks like we could remove this parameter.

File: pdns/ixfrdist.cc [hunk 14]
```diff
@@ -665,7 +665,7 @@ static bool handleAXFR(int fd, const MOADNSParser& mdp) {
 /* Produces an IXFR if one can be made according to the rules in RFC 1995 and
  * creates a SOA or AXFR packet when required by the RFC.
  */
-static bool handleIXFR(int fd, const ComboAddress& destination, const MOADNSParser& mdp, const shared_ptr<SOARecordContent>& clientSOA) {
+static bool handleIXFR(int fd, const ComboAddress& /* destination */, const MOADNSParser& mdp, const shared_ptr<SOARecordContent>& clientSOA) {
```
Comment (rgacogne) [inline pdns/ixfrdist.cc:668]: We can remove this parameter.

File: ext/lmdb-safe/lmdb-safe.cc [hunk 15]
```diff
@@ -16,14 +16,14 @@ static string MDBError(int rc)
   return mdb_strerror(rc);
 }
 
-MDBDbi::MDBDbi(MDB_env* env, MDB_txn* txn, const string_view dbname, int flags)
+MDBDbi::MDBDbi(MDB_env* /* env */, MDB_txn* txn, const string_view dbname, int flags)
```
Comment (rgacogne) [inline ext/lmdb-safe/lmdb-safe.cc:19]: I wonder if we could remove this parameter? It looks that way but I'm not familiar with the LMDB code.

**Review by rgacogne — APPROVED** [review]
Looks great, nice cleanup!

File: ext/lmdb-safe/lmdb-safe.cc (hunk 15 above)
Comment (Habbie) [reply ext/lmdb-safe/lmdb-safe.cc:19]: #12546 is removing tons of dead code from LMDB. Maybe this PR can leave it alone for now :)

**Comment by fredmorcos** [comment]
I also added `-Wall -Wextra` to "Build and test everything" for auth.

File: ext/lmdb-safe/lmdb-safe.cc (hunk 15 above)
Comment (fredmorcos) [reply ext/lmdb-safe/lmdb-safe.cc]: @Habbie [This](https://github.com/PowerDNS/pdns/pull/12549/commits/90d690242f073dd549918456319b679a8a09134e) and [this](https://github.com/PowerDNS/pdns/pull/12549/commits/2ac1eb53ca405c0143e25781b8159755a7502c6e) might come in useful.

File: tasks.py [hunk 16]
```diff
@@ -315,8 +315,8 @@ def ci_auth_configure(c):
     sanitizers = ' '.join('--enable-'+x for x in os.getenv('SANITIZERS').split('+')) if os.getenv('SANITIZERS') != '' else ''
     unittests = ' --enable-unit-tests --enable-backend-unit-tests' if os.getenv('UNIT_TESTS') == 'yes' else ''
     fuzzingtargets = ' --enable-fuzz-targets' if os.getenv('FUZZING_TARGETS') == 'yes' else ''
-    res = c.run('''CFLAGS="-O1 -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int" \
-                   CXXFLAGS="-O1 -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int -Wp,-D_GLIBCXX_ASSERTIONS" \
+    res = c.run('''CFLAGS="-O1 -Wall -Wextra -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int" \
+                   CXXFLAGS="-O1 -Wall -Wextra -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int -Wp,-D_GLIBCXX_ASSERTIONS" \
```
Comment (rgacogne) [inline tasks.py:319]: Aren't these flags already present in `configure.ac`?

**Review by rgacogne — APPROVED** [review]

File: tasks.py (hunk 16 above)
Comment (fredmorcos) [reply tasks.py]: Right. The default FLAGS were also adding `-Wno-unused-parameter` which I now removed and reverted this change.

**Comment by fredmorcos** [comment]
Reminder to merge :-)

File: ext/lmdb-safe/lmdb-safe.cc (hunk 15 above)
Comment (Habbie) [reply ext/lmdb-safe/lmdb-safe.cc]: those now 404, corrected links: [a](https://github.com/PowerDNS/pdns/commit/90d690242f073dd549918456319b679a8a09134e) and [b](https://github.com/PowerDNS/pdns/commit/90d690242f073dd549918456319b679a8a09134e)

Comment (Habbie) [reply ext/lmdb-safe/lmdb-safe.cc]: picked up, thanks!

**Pull request merged**